### PR TITLE
Ctrl C and ctrl v runepages to friends, and more.

### PR DESCRIPTION
**Pull request by robinwithes**
Added a feature where you can select a local runepage and press CTRL c to copy it resulting in a array like: `[Tryndamere, STYLE_8300, STYLE_8100, 8360, 8306, 8321, 8347, 8139, 8135, 5008, 5008, 5002]` Then you or your friends while still having it in your CTRL c buffer can ctrl v it into the local runepage list. This adds a nice way to share runepages with friends.

Also added more runepage list syncs so that you don't need to click the sync button every time you load a new rune page etc.

Example: https://gyazo.com/ec7014ea4622c38b11130d61c679b618

### CODE REVIEW

File: src/main/java/com/stirante/RuneChanger/util/RuneBook.java [hunk 1]
```diff
@@ -32,6 +40,63 @@ public static void refreshLocalRunes(JFXListView<Label> runebookList) {
...
+        String data = null;
+        try {
+            data = (String) transferable.getTransferData(DataFlavor.stringFlavor);
+        } catch (UnsupportedFlavorException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (data == null) {
+            showWarning("Error: Invalid data", "Your CTRL-C buffer contains invalid data", "You can only import runepages exported from Runechanger.");
```
Comment (stirante) [inline src/main/java/com/stirante/RuneChanger/util/RuneBook.java]: I think this should be moved to locale files

File: src/main/java/com/stirante/RuneChanger/model/RunePage.java [hunk 2]
```diff
@@ -9,9 +9,11 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import static com.stirante.RuneChanger.gui.SettingsController.showWarning;
+
 public class RunePage {
-    private final List<Rune> runes = new ArrayList<>(6);
-    private final List<Modifier> modifiers = new ArrayList<>(3);
+    private List<Rune> runes = new ArrayList<>(6);
+    private List<Modifier> modifiers = new ArrayList<>(3);
```
Comment (stirante) [inline src/main/java/com/stirante/RuneChanger/model/RunePage.java]: Those lists can still be final. You don't have to create new ArrayList.

File: src/main/java/com/stirante/RuneChanger/util/RuneBook.java [hunk 3]
```diff
@@ -32,6 +40,63 @@ public static void refreshLocalRunes(JFXListView<Label> runebookList) {
...
+            return;
+        }
+
+        if (!runePage.verify()) {
+            showWarning("Invalid Runepage", "Error: Invalid Runepage", "Please copy a valid runepage to continue.");
+        }
+
+        StringSelection selection = new StringSelection(runePage.exportRunePage().toString());
+        clipboard.setContents(selection, selection);
+        showInfoAlert("Succesfully copied Rune Data", "You succesfully copied the selected Rune Data!", "");
```
Comment (stirante) [inline src/main/java/com/stirante/RuneChanger/util/RuneBook.java]: I think this should be moved to locale files

File: src/main/java/com/stirante/RuneChanger/util/RuneBook.java [hunk 4]
```diff
@@ -32,6 +40,63 @@ public static void refreshLocalRunes(JFXListView<Label> runebookList) {
...
+            return;
+        }
+        List<String> list = stringToList(data);
+        RunePage page = new RunePage();
+        if (!page.importRunePage(list)) {
+            showWarning("Invalid Runepage", "Invalid Runepage", "This runepage does not have the required information to be imported!");
+            return;
+        }
+        if ((!localList.getItems().filtered(label -> label.getText().equalsIgnoreCase(page.getName())).isEmpty())) {
+            showWarning("Invalid Runepage", "Duplicate Name Error", "Another runepage with the same name already exists");
```
Comment (stirante) [inline src/main/java/com/stirante/RuneChanger/util/RuneBook.java]: I think this should be moved to locale files

File: src/main/java/com/stirante/RuneChanger/util/RuneBook.java [hunk 5]
```diff
@@ -32,6 +40,63 @@ public static void refreshLocalRunes(JFXListView<Label> runebookList) {
...
+            e.printStackTrace();
+        }
+        if (data == null) {
+            showWarning("Error: Invalid data", "Your CTRL-C buffer contains invalid data", "You can only import runepages exported from Runechanger.");
+            return;
+        }
+        List<String> list = stringToList(data);
+        RunePage page = new RunePage();
+        if (!page.importRunePage(list)) {
+            showWarning("Invalid Runepage", "Invalid Runepage", "This runepage does not have the required information to be imported!");
```
Comment (stirante) [inline src/main/java/com/stirante/RuneChanger/util/RuneBook.java]: I think this should be moved to locale files

File: src/main/java/com/stirante/RuneChanger/util/RuneBook.java [hunk 6]
```diff
@@ -32,6 +40,63 @@ public static void refreshLocalRunes(JFXListView<Label> runebookList) {
...
+            showWarning(LangHelper.getLang().getString("no_page_with_name"), String.format(LangHelper.getLang()
+                    .getString("no_page_with_name_message"), selectedListView
+                    .getFocusModel()
+                    .getFocusedItem()
+                    .getText()), null);
+            return;
+        }
+
+        if (!runePage.verify()) {
+            showWarning("Invalid Runepage", "Error: Invalid Runepage", "Please copy a valid runepage to continue.");
```
Comment (stirante) [inline src/main/java/com/stirante/RuneChanger/util/RuneBook.java]: I think this should be moved to locale files

File: src/main/java/com/stirante/RuneChanger/model/RunePage.java (hunk 2 above)
Comment (robinwithes) [reply src/main/java/com/stirante/RuneChanger/model/RunePage.java]: I'm pretty sure those can't be final, otherwise i'm not sure what you're hinting at.

**Review by stirante — APPROVED** [review]

**Pull request merged**